lib/kafka/consumer_interceptors.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Kafka
+  # Holds a list of Kafka::ConsumerInterceptor
+  # and wraps calls to a chain of custom consumer interceptors.
+  class ConsumerInterceptors
+    def initialize(interceptors:, logger:)
+      @interceptors = interceptors
+      @logger = TaggedLogger.new(logger)
+    end
+
+    # This is called from Kafka::Consumer once the batches are fetched.
+    #
+    # @param batches [Array<Kafka::FetchedMessage>]
+    # @return [Array<Kafka::FetchedMessage>] intercepted batches.
+    def on_consume(batches:)
+      intercepted_batches = batches
+      @interceptors.each do |interceptor|
+        begin
+          intercepted_batches = interceptor.on_consume(batches: intercepted_batches)
+        rescue Exception => e
+          @logger.warn "Error executing interceptor on_consume #{e.class}: #{e.message}\n#{e.backtrace.join("\n")}"
```

Comment:
is it necessary to specify all of `e` `class,message,backtrace`?
or would logging just `#{e}` would be enough?

Reply:
I did it like that because https://github.com/zendesk/ruby-kafka/blob/v1.1.0/lib/kafka/async_producer.rb#L239-L240 is also done that way